pages/404.html

```diff
@@ -0,0 +1,14 @@
+<#include "incl/macros.html">
+<script>
+  const path = window.location.pathname;
+  const stdlibRegex = /^\/(stdlib|library)\//;
+  const refmanRegex = /^\/distrib\/(.*)\/refman\//;
+  if (stdlibRegex.test(path)) {
+    window.location = path.replace(stdlibRegex, "/doc/<#CURRENTVERSIONTAG>/stdlib/");
+  } else if (refmanRegex.test(path)) {
+    const version = path.match(refmanRegex)[1];
+    window.location = path.replace(refmanRegex, "/doc/" + version + "/refman/");
```

Comment:
The case of `distrib/current/refman` will need to be handled specially (using the `#CURRENTVERSIONTAG`). BTW, this case also exists for the stdlib doc (as in `/distrib/current/stdlib` or `/distrib/<version>/stdlib`).

Reply:
The distrib/current/refman is already handled by meta refresh redirect (https://htk-coq-website-fix.github.io/distrib/current/refman)

Reply:
The same goes for /distrib/current/stdlib

Reply:
This actually does not handle sub-paths. Here is an example of broken URL: https://htk-coq-website-fix.github.io/doc/current/refman/addendum/type-classes.html (already broken before this PR of course).